### PR TITLE
Added SSE Parameter - 0.4

### PR DESCRIPTION
**Pull request by jrs526**
Enables server-side encryption by adding a parameter, `sse`, which takes a string denoting the encryption algorithm.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -130,6 +131,14 @@ func command(s S3) *exec.Cmd {
...
 		s.Region,
+		"--sse",
+		s.SSE,
+	}
+
+	// if sse is empty, remove from the
+	// above arguments.
+	if len(s.SSE) == 0 {
+		args = append(args[:9], args[9+2:]...)
 	}
```
Comment (bradrydzewski) [inline main.go]: a slightly better approach might be:

```
if s.SSSE != "" {
    args = append(args, "--sse")
    args = append(args, s.SSE)
}
```

Comment (jrs526) [inline main.go]: Agreed.  I'm making the change now.

File: main.go [hunk 2]
```diff
@@ -125,17 +126,18 @@ func command(s S3) *exec.Cmd {
...
-	// if not recursive, remove from the
-	// above arguments.
-	if !s.Recursive {
-		args = append(args[:4], args[4+1:]...)
+	if s.SSE != "" {
+		args = append(args, "--sse", s.SSE)
+	}
+
+	if s.Recursive {
+		args = append(args, "--recursive")
```
Comment (jrs526) [inline main.go]: I went ahead and changed the way the recursive parameter was handled to match the rest of the conditional parameters.

**Comment by tboerger** [comment]
LGTM, but should also be ported to 0.5

**Comment by jrs526** [comment]
I've got a branch for porting to 0.5, I just haven't had a chance to test it yet.

**Pull request closed without merging**